### ISSUE TITLE
Merging documents drops anchor, keeps alias

### ISSUE THREAD

**Issue by hobti01** [mikefarah/yq#656]
**Describe the bug**
Merging two documents drops anchors (haha) while keeping the aliases - creating an invalid document.

Version of yq: 4.3.1
Operating system: mac
Installed via: homebrew

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)
data1.yml:
```yaml
test1: bar
```

data2.yml:
```yaml
test2: &anchor "foo"
alias: *anchor
```

**Command**
The command you ran:
```
yq eval-all "select(fileIndex == 0) * select(fileIndex == 1)" data1.yml data2.yml
```

**Actual behavior**

```yaml
test1: bar
test2: "foo"
alias: *anchor
```

**Expected behavior**

```yaml
test1: bar
test2: &anchor "foo"
alias: *anchor
```

**Additional context**
* Order of the documents does not matter, same result.
* All yaml anchors are removed.
* I'd prefer not to explode the references but please let me know if that's the only solution.

**Comment by mikefarah** [comment]
yep - confirmed the bug, will fix in the next release

**Comment by mikefarah** [comment]
Fixed in 4.3.2
